build-tools/src/main/kotlin/org/jetbrains/kotlin/ExecutorService.kt

```diff
@@ -311,3 +320,233 @@ private fun sshExecutor(project: Project) : ExecutorService = object : ExecutorS
+        val out = ByteArrayOutputStream()
+
+        val result = project.exec {
+            it.workingDir = xcProject.toFile()
+            it.commandLine = listOf(idb, "debugserver", "start", "--udid", udid, bundleId)
+            it.standardOutput = out
+            it.errorOutput = out
+        }
+        check(result.exitValue == 0) { "Failed to start debug server: $out" }
+        return out.toString()
+    }
+}
+
+fun KonanTestExecutable.xcodeBuild() {
+    this.doBeforeRun = Action {
+        val signIdentity = project.findProperty("sign_identity") as? String ?: "iPhone Developer"
+        val developmentTeam = project.findProperty("development_team") as? String
+        val xcProject = Paths.get(project.testOutputRoot, "launcher")
+
+        val shellScript: String = // language=Bash
+                mutableListOf("""
+                        # Copy executable to the build dir.
+                        COPY_TO="${"$"}TARGET_BUILD_DIR/${"$"}EXECUTABLE_PATH"
+                        cp "${"$"}PROJECT_DIR/KonanTestLauncher/build/${"$"}TARGET_NAME.kexe" "${"$"}COPY_TO"
+                        # copy dSYM if it exists
+                        DSYM_DIR="${"$"}PROJECT_DIR/KonanTestLauncher/build/${"$"}TARGET_NAME.kexe.dSYM"
+                        if [ -d "${"$"}DSYM_DIR" ]; then
+                            cp -r "${"$"}DSYM_DIR" "${"$"}TARGET_BUILD_DIR/${"$"}EXECUTABLE_FOLDER_PATH/"
+                        fi
+                    """.trimIndent()).also {
+                    when (this) {
+                        is FrameworkTest -> {
+                            // Create a Frameworks folder inside the build dir.
+                            it += "mkdir -p \"\$TARGET_BUILD_DIR/\$FRAMEWORKS_FOLDER_PATH\""
+                            // Copy each framework to the Frameworks dir.
+                            it += frameworkNames.map { name ->
+                                "cp -r \"$testOutput/$testName/${project.testTarget.name}/$name.framework\" " +
+                                        "\"\$TARGET_BUILD_DIR/\$FRAMEWORKS_FOLDER_PATH/$name.framework\""
+                            }
+                        }
+                    }
+                }.joinToString(separator = "\\n") { it.replace("\"", "\\\"") }
+
+        // Copy template xcode project.
+        project.file("iosLauncher").copyRecursively(xcProject.toFile(), overwrite = true)
+
+        xcProject.resolve("KonanTestLauncher.xcodeproj/project.pbxproj")
+                .toFile()
+                .apply {
+                    val text = readLines().joinToString("\n") {
+                        when {
+                            it.contains("CODE_SIGN_IDENTITY") ->
+                                it.replaceAfter("= ", "\"$signIdentity\";")
+                            it.contains("DEVELOPMENT_TEAM") || it.contains("DevelopmentTeam") ->
+                                it.replaceAfter("= ", "$developmentTeam;")
```

Comment:
If `development_team` isn't set, `null` is written here. May be add a check separate check that `development_team` is specified?